### PR TITLE
feat: add run uuid to cli

### PR DESCRIPTION
**Pull request by hugorut**
* Adds uuid to run context which is populated to ApiClient
* Remove `X-Trace-Id` header which was unused and replace with `X-Infracost-Trace-Id` populated with run context uuid

See https://github.com/infracost/cloud-pricing-api/pull/137 for pricing api implementation
See https://github.com/infracost/dashboard/pull/26 for dashboard implementation

### CODE REVIEW

**Comment by hugorut** [comment]
Tested with both `cloud-pricing-api` and `hosted-cloud-pricing-api` and going through to mixpanel fine. Sentry alerts also capturing request header `X-Infracost-Trace-Id` when errors caught.

File: internal/apiclient/pricing.go [hunk 1]
```diff
@@ -29,8 +29,8 @@ type PriceQueryResult struct {
 	Result gjson.Result
 }
 
-func NewPricingAPIClient(cfg *config.Config) *PricingAPIClient {
-	currency := cfg.Currency
+func NewPricingAPIClient(cfg *config.RunContext) *PricingAPIClient {
```
Comment (vdmgolub) [inline internal/apiclient/pricing.go:32]: No biggy, but it seems like `ctx` would be a more descriptive variable name with the updated type. What do you think?
```suggestion
func NewPricingAPIClient(ctx *config.RunContext) *PricingAPIClient {
```

File: internal/prices/prices.go [hunk 2]
```diff
@@ -12,7 +12,7 @@ import (
 	"github.com/tidwall/gjson"
 )
 
-func PopulatePrices(cfg *config.Config, project *schema.Project) error {
+func PopulatePrices(cfg *config.RunContext, project *schema.Project) error {
```
Comment (vdmgolub) [inline internal/prices/prices.go:15]: Same comment about the name.
```suggestion
func PopulatePrices(ctx *config.RunContext, project *schema.Project) error {
```

**Review by vdmgolub — APPROVED** [review]
Looks great 👍 I've added a couple minor comments about arguments naming, please take a look.

File: internal/apiclient/pricing.go (hunk 1 above)
Comment (hugorut) [reply internal/apiclient/pricing.go:32]: good shout

**Pull request merged**